### PR TITLE
tests: add test for module publish signal to reset cache

### PR DESCRIPTION
**Pull request by goapunk**
**Describe your changes**
Briefly explain what you did and provide context for a clearer understanding.

**Tasks**
- [ ] PR name contains story or task reference
- [ ] Steps to recreate and test the changes
- [ ] Documentation (docs and inline)
- [x] Tests (including n+1 and django_assert_num_queries where applicable)
- [ ] Changelog

### CODE REVIEW

File: tests/projects/test_tasks_caching.py [hunk 1]
```diff
@@ -412,3 +414,110 @@ def test_new_phase_resets_cache(
...
+        module__project=past_project,
+    )
+
+    module1 = phase_1.module
+    module2 = phase_2.module
+    module1.is_draft = True
+    module1.save()
+    module2.is_draft = True
+    module2.save()
+
```
Comment (m4ra) [inline tests/projects/test_tasks_caching.py:488]: here good occasion to check if cache is empty for these projects now.

File: tests/projects/test_tasks_caching.py [hunk 2]
```diff
@@ -412,3 +414,110 @@ def test_new_phase_resets_cache(
...
+        module__project=future_project,
+    )
+    phase_2 = phase_factory(
+        start_date=last_week,
+        end_date=next_week,
+        module__project=past_project,
+    )
+
+    module1 = phase_1.module
+    module2 = phase_2.module
```
Comment (m4ra) [inline tests/projects/test_tasks_caching.py:483]: worth testing the past projct with a new future module, so we have the multimodule covered too.

**Review by m4ra — COMMENTED** [review]
Great, just few little proposals to have some extra states covered.

File: tests/projects/test_tasks_caching.py (hunk 1 above)
Comment (goapunk) [reply tests/projects/test_tasks_caching.py:502]: added

File: tests/projects/test_tasks_caching.py (hunk 2 above)
Comment (goapunk) [reply tests/projects/test_tasks_caching.py]: added a future module to past project instead of active

File: tests/projects/test_tasks_caching.py [hunk 3]
```diff
@@ -412,3 +414,125 @@ def test_new_phase_resets_cache(
...
+        module__project=future_project,
+    )
+    # turn past project into future project
+    phase_future = phase_factory(
+        start_date=next_week,
+        end_date=next_week + timedelta(days=7),
+        module__project=past_project,
+    )
+
+    module1 = phase_active.module
```
Comment (m4ra) [inline tests/projects/test_tasks_caching.py:484]: so this is a new module? didn't realise it. so every phase factory autocreates a new module? @goapunk

Comment (goapunk) [reply tests/projects/test_tasks_caching.py:484]: unless you specify an existing module it creates a new module, yes

**Review by m4ra — APPROVED** [review]

**Pull request merged**